datasets/amazon_reviews_multi/amazon_reviews_multi.py

```diff
@@ -0,0 +1,132 @@
+"""
+
+_LANGUAGES = {
+    "de": "German",
+    "en": "English",
+    "es": "Spanish",
+    "fr": "French",
+    "ja": "Japanese",
+    "zh": "Chinese",
+}
+_ALL_LANGUAGES = "all_languages"
+_VERSION = "1.0.0"
+_HOMEPAGE_URL = "https://registry.opendata.aws/amazon-reviews-ml/"
+_DOWNLOAD_URL = "https://amazon-reviews-ml.s3-us-west-2.amazonaws.com/json/{split}/dataset_{lang}_{split}.json"
+
+
+class AmazonReviewsMultiConfig(datasets.BuilderConfig):
+    """BuilderConfig for AmazonReviewsMultiConfig."""
+
+    def __init__(self, languages=None, **kwargs):
+        super(AmazonReviewsMultiConfig, self).__init__(version=datasets.Version(_VERSION, ""), **kwargs),
+        self.languages = languages
+
+
+class AmazonReviewsMulti(datasets.GeneratorBasedBuilder):
+    """The Multilingual Amazon Reviews Corpus"""
+
+    BUILDER_CONFIGS = [
+        AmazonReviewsMultiConfig(
+            name=_ALL_LANGUAGES,
+            languages=_LANGUAGES,
+            description="A collection of Amazon reviews specifically designed to aid research in multilingual text classification.",
+        )
+    ] + [
+        AmazonReviewsMultiConfig(
+            name=lang,
+            languages=[lang],
+            description=f"{_LANGUAGES[lang]} examples from a collection of Amazon reviews specifically designed to aid research in multilingual text classification",
+        )
+        for lang in _LANGUAGES
+    ]
+    BUILDER_CONFIG_CLASS = AmazonReviewsMultiConfig
+    DEFAULT_CONFIG_NAME = _ALL_LANGUAGES
+
+    def _info(self):
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=datasets.Features(
+                {
+                    "review_id": datasets.Value("string"),
+                    "product_id": datasets.Value("string"),
+                    "reviewer_id": datasets.Value("string"),
+                    "stars": datasets.Value("int32"),
+                    "review_body": datasets.Value("string"),
+                    "review_title": datasets.Value("string"),
+                    "language": datasets.Value("string"),
+                    "product_category": datasets.Value("string"),
+                }
+            ),
+            supervised_keys=None,
+            license=_LICENSE,
+            homepage=_HOMEPAGE_URL,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        train_urls = [_DOWNLOAD_URL.format(split="train", lang=lang) for lang in self.config.languages]
+        dev_urls = [_DOWNLOAD_URL.format(split="dev", lang=lang) for lang in self.config.languages]
+        test_urls = [_DOWNLOAD_URL.format(split="test", lang=lang) for lang in self.config.languages]
+
+        train_paths = [dl_manager.download_and_extract(url) for url in train_urls]
+        dev_paths = [dl_manager.download_and_extract(url) for url in dev_urls]
+        test_paths = [dl_manager.download_and_extract(url) for url in test_urls]
```

Comment:
Just for information you can use 
```python
train_urls = [_DOWNLOAD_URL.format(split="train", lang=lang) for lang in self.config.languages]
train_paths = dl_manager.download_and_extract(train_urls)
```
directly and ouput a list of paths.
It can be quite convenient sometimes and it allows parallel downloads if there are many many files that need to be downloaded.

(here you can leave it this way though)

Reply:
Oh no that's way better, thanks